repository/src/main/java/org/apache/atlas/repository/graph/GraphHelper.java

```diff
@@ -359,18 +360,32 @@ public static boolean getRestrictPropagationThroughHierarchy(AtlasVertex classif
         return getRestrictPropagation(classificationVertex,CLASSIFICATION_VERTEX_RESTRICT_PROPAGATE_THROUGH_HIERARCHY);
     }
 
-    public static AtlasVertex getClassificationVertex(AtlasVertex entityVertex, String classificationName) {
+    public void repairTagVertex(AtlasEdge edge, AtlasVertex classificationVertex) {
+        LOG.info("Repairing corrupt tag-vertex");
+        removeEdge(edge);
+        removeVertex(classificationVertex);
+    }
+
+    public static AtlasVertex getClassificationVertex(GraphHelper graphHelper, AtlasVertex entityVertex, String classificationName) {
         AtlasVertex ret   = null;
         Iterable    edges = entityVertex.query().direction(AtlasEdgeDirection.OUT).label(CLASSIFICATION_LABEL)
                                                 .has(CLASSIFICATION_EDGE_IS_PROPAGATED_PROPERTY_KEY, false)
                                                 .has(CLASSIFICATION_EDGE_NAME_PROPERTY_KEY, classificationName).edges();
+
         if (edges != null) {
             Iterator<AtlasEdge> iterator = edges.iterator();
 
-            if (iterator.hasNext()) {
+            while (iterator.hasNext()) {
                 AtlasEdge edge = iterator.next();
-
-                ret = (edge != null) ? edge.getInVertex() : null;
+                if(Objects.nonNull(edge))
+                {
+                    AtlasVertex classificationVertex = edge.getInVertex();
+                    if(Objects.nonNull(classificationVertex) && StringUtils.isNotEmpty(classificationVertex.getProperty(TYPE_NAME_PROPERTY_KEY, String.class))) {
```

Comment:
Make sure this is not throwing error

Reply:
regarding line 383? if yes, then yeah i checked, no error. I even dropped the typeName field and then it simply returned null which was handled by the check